Userland/Utilities/touch.cpp

```diff
@@ -21,31 +38,227 @@ static bool file_exists(String const& path)
         if (errno == ENOENT)
             return false;
     }
-    if (rc == 0) {
+    if (rc == 0)
         return true;
+    err("failed to get attributes of '{}': {}", path, strerror(errno));
+}
+
+inline static bool validate_timestamp(unsigned year, unsigned month, unsigned day, unsigned hour, unsigned minute, unsigned second)
+{
+    return (year >= 1970) && (month >= 1 && month <= 12) && (day >= 1 && day <= (unsigned)days_in_month(year, month)) && (hour <= 23) && (minute <= 59) && (second <= 59);
+}
+
+static void parse_time(StringView strtime, timespec& atime, timespec& mtime)
+{
+    // Parse [[CC]YY]MMDDhhmm[.SS] format, where brackets signify optional
+    // parameters.
+    if (strtime.length() < 8)
+        err("invalid time format '{}' -- too short", strtime);
+    else if (strtime.length() > 15)
+        err("invalid time format '{}' -- too long", strtime);
+
+    Vector<u8> params;
+    GenericLexer lexer(strtime);
+    unsigned year, month, day, hour, minute, second;
+
+    auto lex_number = [&](size_t n) {
```

Comment:
This doesn't need a size parameter, you always use it with and check for 2 number characters.